app/controllers/articles_controller.rb

```diff
@@ -0,0 +1,60 @@
+class ArticlesController < ApplicationController
+  before_action :find_article, except: %i[index new create]
+
+  http_basic_authenticate_with name: 'user', password: 'secret', except: %i[index show]
+
+  def index
+    @articles = Article.all
+  end
+
+  def show
+    @article 
```

Comment:
Is it necessary to return the `@article` variable?

Reply:
I tested the method without return the `@article` variable and works fine, but I have a question. Should I delete the `show`method ?

Reply:
Yes, you can delete it. That's because you're using `resources :articles` in the routes file. You can read about this [here](https://guides.rubyonrails.org/layouts_and_rendering.html#rendering-by-default-convention-over-configuration-in-action).

Reply:
Thank you for the explanation!